lib/devices.go

```diff
@@ -244,23 +264,28 @@ func (d *Device) treeNode() TreeNode {
 	}
 }
 
+func flatten(path []int) string {
+	s := ""
+	for _, p := range path {
+		s += fmt.Sprintf("%04d-", p)
+	}
+
+	return s
+}
+
 // sortDevices sorts devices consistently by their path
 func sortDevices(devices []Device) []Device {
 	sort.Slice(devices, func(i, j int) bool {
-		flatten := func(path []int) string {
-			s := ""
-			for _, p := range path {
-				s += fmt.Sprintf("%04d-", p)
-			}
-			return s
-		}
-
 		return flatten(devices[i].Path) < flatten(devices[j].Path)
 	})
 
 	return devices
 }
 
+func addErrorLog(text string, logtime time.Time) {
+	logs = append(logs, Log{Time: logtime, Text: text})
```

Comment:
You should want an `error` state here, so you can put `state` in a `class` on the frontend and color it appropriately e.g. with orange.